src/SubscriptionItem.php

```diff
@@ -228,11 +279,92 @@ public function reportUsage($quantity = 1, $timestamp = null)
      */
     public function usageRecords($options = [])
     {
-        return new Collection($this->subscription->owner->stripe()->subscriptionItems->allUsageRecordSummaries(
-            $this->stripe_id, $options
+        $meterId = $this->meter_id;
+
+        if (! $meterId) {
+            // Get the price to determine the meter
+            $stripePrice = $this->subscription->owner->stripe()->prices->retrieve($this->stripe_price);
+
+            if (! isset($stripePrice->recurring->meter)) {
+                throw new \InvalidArgumentException('Price must have a meter to get usage records. Legacy usage records are no longer supported.');
+            }
+
+            $meterId = $stripePrice->recurring->meter;
+
+            $this->forceFill(['meter_id' => $meterId])->save();
+        }
+
+        // Default time range - current billing period
+        $defaultOptions = [
+            'start_time' => $this->currentPeriodStart()?->getTimestamp() ?? 1,
+            'end_time' => time(),
+            'customer' => $this->subscription->owner->stripeId(),
+        ];
+
+        return new Collection($this->subscription->owner->stripe()->billing->meters->allEventSummaries(
+            $meterId,
+            array_merge($defaultOptions, $options)
         )->data);
     }
 
+    /**
+     * Get the current period start date for this subscription item.
+     *
+     * @param  string|null  $timezone
+     * @return \Carbon\Carbon|null
+     */
+    public function currentPeriodStart($timezone = null)
+    {
+        $stripeItem = $this->asStripeSubscriptionItem();
+
+        if (! isset($stripeItem->current_period_start)) {
+            return null;
+        }
+
+        $date = Carbon::createFromTimestamp($stripeItem->current_period_start);
```

Comment:
@driesvints Do you have any opinion whether to use `Carbon`, `CarbonImmutable` or use `Illuminate\Support\Date` so application developers can decide whether to use immutable or not?

Reply:
No hard opinion but I'd say continue following what was used already. Personally I'd use `CarbonImmutable` but I don't know of the implications on the package.